lighting/raymarch/volume.glsl

```diff
@@ -1,103 +1,137 @@
 #include "map.glsl"
-#include "normal.glsl"
+#include "../light/attenuation.glsl"
+#include "../../generative/random.glsl"
+#include "../../math/const.glsl"
+#include "../medium/new.glsl"
 
 /*
-contributors:  Inigo Quiles
-description: Default raymarching renderer
-use: <vec4> raymarchVolume( in <float3> rayOriging, in <float3> rayDirection, in <float3> cameraForward,
-    out <float3> eyeDepth, out <float3> worldPosition, out <float3> worldNormal ) 
+contributors:  Shadi El Hajj
+description: Default raymarching renderer. Based on Sébastien Hillaire's paper "Physically Based Sky, Atmosphere & Cloud Rendering in Frostbite"
+use: <vec4> raymarchVolume(<vec3> rayOrigin, <vec3> rayDirection, <vec2> st, float minDist, <vec3> background) 
 options:
-    - RAYMARCH_BACKGROUND vec3(0.0)
-    - RAYMARCH_AMBIENT vec3(1.0)
-    - LIGHT_COLOR     vec3(0.5)
-    - LIGHT_POSITION  vec3(0.0, 10.0, -50.0)
+    - RAYMARCH_VOLUME_SAMPLES       64
+    - RAYMARCH_VOLUME_SAMPLES_LIGHT 32
+    - RAYMARCH_VOLUME_MAP_FNC       raymarchVolumeMap
+    - RAYMARCH_VOLUMETRIC_SHADOWS
+    - RAYMARCH_VOLUME_DITHER        0.1
+    - RAYMARCH_ENERGY_CONSERVING
+    - LIGHT_COLOR                   vec3(0.5, 0.5, 0.5)
+    - LIGHT_INTENSITY               1.0
+    - LIGHT_POSITION
+    - LIGHT_DIRECTION
 examples:
     - /shaders/lighting_raymarching_volume.frag
+license: MIT License (MIT) Copyright (c) 2024 Shadi EL Hajj
 */
 
 #ifndef LIGHT_COLOR
 #if defined(GLSLVIEWER)
 #define LIGHT_COLOR u_lightColor
 #else
-#define LIGHT_COLOR vec3(0.5)
+#define LIGHT_COLOR vec3(0.5, 0.5, 0.5)
 #endif
 #endif
 
-#ifndef RAYMARCH_BACKGROUND
-#define RAYMARCH_BACKGROUND vec3(0.0)
+#ifndef LIGHT_INTENSITY
+#define LIGHT_INTENSITY 1.0
 #endif
 
-#ifndef RAYMARCH_SAMPLES
-#define RAYMARCH_SAMPLES 64
+#ifndef RAYMARCH_VOLUME_SAMPLES
+#define RAYMARCH_VOLUME_SAMPLES 64
 #endif
 
-#ifndef RAYMARCH_MIN_DIST
-#define RAYMARCH_MIN_DIST 1.0
+#ifndef RAYMARCH_VOLUME_SAMPLES_LIGHT
+#define RAYMARCH_VOLUME_SAMPLES_LIGHT 32
 #endif
 
-#ifndef RAYMARCH_MAX_DIST
-#define RAYMARCH_MAX_DIST 10.0
+#ifndef RAYMARCH_VOLUME_MAP_FNC
+#define RAYMARCH_VOLUME_MAP_FNC raymarchVolumeMap
 #endif
 
-#ifndef RAYMARCH_MAP_FNC
-#define RAYMARCH_MAP_FNC(POS) raymarchMap(POS)
+#ifndef RAYMARCH_VOLUME_DITHER
+#define RAYMARCH_VOLUME_DITHER 0.1
 #endif
 
 #ifndef FNC_RAYMARCH_VOLUMERENDER
```

Comment:
We could change the name of this one 

```
#if !defined(FNC_RAYMARCH_VOLUME) && defined(RAYMARCH_VOLUME)
#define FNC_RAYMARCH_VOLUME
```

Reply:
@patriciogonzalezvivo  Even better!! I'll amend.

Reply:
Done!